controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/TruncateStreamTask.java

```diff
@@ -69,19 +70,24 @@ public TruncateStreamTask(final StreamMetadataTasks streamMetadataTasks,
         final OperationContext context = streamMetadataStore.createStreamContext(scope, stream, requestId);
 
         return streamMetadataStore.getVersionedState(scope, stream, context, executor)
-                .thenCompose(versionedState -> streamMetadataStore.getTruncationRecord(scope, stream, context, executor)
-                        .thenCompose(versionedMetadata -> {
-                            if (!versionedMetadata.getObject().isUpdating()) {
-                                if (versionedState.getObject().equals(State.TRUNCATING)) {
-                                    return Futures.toVoid(streamMetadataStore.updateVersionedState(scope, stream, State.ACTIVE,
-                                            versionedState, context, executor));
+                .thenCompose(versionedState -> {
+                    if (versionedState.getObject().equals(State.SEALED)) {
+                        throw new StoreException.IllegalStreamStateException("ERROR: Attempting to truncate sealed stream " + stream, null);
```

Comment:
Instead of `new StoreException.IllegalStreamStateException(. . .)` please use StoreException.create() as done elsewhere...